solutions/usaco-1064.mdx

```diff
@@ -108,4 +108,105 @@ int main() {
 ```
 
 </CPPSection>
+	
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class StuckInARut {
+	static int[] xCoordinates;
+	static int[] yCoordinates;
+
+	// sorts eastward cows by their y value
+	static class EastComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			return yCoordinates[first] - yCoordinates[second];
+		}
+	}
+
+	// sorts northward cows by their x value
+	static class NorthComparator implements Comparator<Integer> {
+		public int compare (Integer first, Integer second) {
+			return xCoordinates[first] - xCoordinates[second];
+		}
+	}
```

Comment:
lambdas exist

Reply:
bump

Reply:
I did this so that they wouldn't be used

Reply:
wha

Reply:
unaddressed